From 54132324878f2d47c134c201fd27206cffcfcfea Mon Sep 17 00:00:00 2001
From: Jan Palus <jpalus@fastmail.com>
Date: Thu, 22 Jun 2023 12:58:44 +0200
Subject: [PATCH] Fix compatibility with tinydb < 4.0.0

tinydb 4.0.0 extracted Document/Table classes from database to table
module
---
 etm/model.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/etm/model.py b/etm/model.py
index ee8badf..a1479f9 100755
--- a/etm/model.py
+++ b/etm/model.py
@@ -41,7 +41,11 @@ import sys
 import re
 
 from tinydb import __version__ as tinydb_version
-from tinydb.table import Document
+from packaging.version import parse as parse_version
+if parse_version(tinydb_version) >= parse_version("4.0.0"):
+    from tinydb.table import Document
+else:
+    from tinydb.database import Document
 
 from jinja2 import Template
 from jinja2 import __version__ as jinja2_version
-- 
2.41.0

